packages/core/src/modules/routing/services/MediationRecipientService.ts

```diff
@@ -247,19 +251,43 @@ export class MediationRecipientService {
     if (!mediationRecord) return routing
 
     // new did has been created and mediator needs to be updated with the public key.
-    mediationRecord = await this.keylistUpdateAndAwait(
-      agentContext,
-      mediationRecord,
-      routing.recipientKey.publicKeyBase58
-    )
+    mediationRecord = await this.keylistUpdateAndAwait(agentContext, mediationRecord, [
+      {
+        recipientKey: routing.recipientKey,
+        action: KeylistUpdateAction.add,
+      },
+    ])
 
     return {
       ...routing,
+      mediatorId: mediationRecord.id,
       endpoints: mediationRecord.endpoint ? [mediationRecord.endpoint] : routing.endpoints,
       routingKeys: mediationRecord.routingKeys.map((key) => Key.fromPublicKeyBase58(key, KeyType.Ed25519)),
     }
   }
 
+  public async removeMediationRouting(
+    agentContext: AgentContext,
+    { recipientKeys, mediatorId }: RemoveRoutingOptions
```

Comment:
On the service level we usually use records over ids, not sure if using a record is better here though